spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitGatewaySupport.java

```diff
@@ -72,28 +73,28 @@ protected RabbitTemplate createRabbitTemplate(ConnectionFactory connectionFactor
 	 * @return The Rabbit ConnectionFactory used by the gateway.
 	 */
 	public final ConnectionFactory getConnectionFactory() {
-		return (this.rabbitTemplate != null ? this.rabbitTemplate.getConnectionFactory() : null);
+		return (this.rabbitOperations != null ? this.rabbitOperations.getConnectionFactory() : null);
 	}
 
 	/**
-	 * Set the RabbitTemplate for the gateway.
-	 * @param rabbitTemplate The Rabbit template.
+	 * Set the {@link RabbitOperations} for the gateway.
+	 * @param rabbitOperations The Rabbit operations.
 	 * @see #setConnectionFactory(org.springframework.amqp.rabbit.connection.ConnectionFactory)
 	 */
-	public final void setRabbitTemplate(RabbitTemplate rabbitTemplate) {
-		this.rabbitTemplate = rabbitTemplate;
+	public final void setRabbitTemplate(RabbitTemplate rabbitOperations) {
```

Comment:
??? For version 1.6 you have `@Deprecated` this method. 
And I don't see here new `setRabbitOperations()`